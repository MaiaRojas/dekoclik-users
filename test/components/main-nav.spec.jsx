import React from 'react';
import { shallow } from 'enzyme';
import MainNav from '../../src/components/main-nav';


describe.skip('<MainNav />', () => {

  it('should warn when missing props: auth, match, history, firebase', () => {
    const spy = jest.spyOn(console, 'error').mockImplementation(jest.fn());

    shallow(<MainNav />);

    expect(spy.mock.calls.length).toBe(4);
    expect(spy.mock.calls[0]).toMatchSnapshot();
    expect(spy.mock.calls[0]).toMatchSnapshot();
    expect(spy.mock.calls[0]).toMatchSnapshot();
    expect(spy.mock.calls[0]).toMatchSnapshot();

    spy.mockReset();
    spy.mockRestore();
  });

  it('should warn when missing nested props: history.push, firebase.logout', () => {
    const spy = jest.spyOn(console, 'error').mockImplementation(() => null);
    shallow((
      <MainNav
        auth={{ displayName: 'Ada Lovelace', email: 'ada@gmail.com' }}
        match={{ path: '/' }}
        history={{}}
        firebase={{}}
      />
    ));

    expect(spy.mock.calls.length).toBe(2);
    expect(spy.mock.calls[0]).toMatchSnapshot();
    expect(spy.mock.calls[1]).toMatchSnapshot();

    spy.mockReset();
    spy.mockRestore();
  });

  it('should render a <LeftDrawer> with a <List>', () => {
    const component = shallow((
      <MainNav
        auth={{ displayName: 'Ada Lovelace', email: 'ada@gmail.com' }}
        match={{ path: '/' }}
        history={{ push: () => {} }}
        firebase={{ logout: () => {} }}
      />
    )).dive();

    const el = component.getElement();
    expect(el.type.Naked.name).toBe('LeftDrawer');
    expect(el.type.displayName).toBe('Connect(WithStyles(LeftDrawer))');
    expect(el.props.children.type.Naked.name).toBe('List');
    expect(el.props.children.type.displayName).toBe('WithStyles(List)');
  });

  it('should render a <List> with array of <ListItem>s or <Divider>s', () => {
    const component = shallow((
      <MainNav
        auth={{ displayName: 'Ada Lovelace', email: 'ada@gmail.com' }}
        match={{ path: '/' }}
        history={{ push: () => {} }}
        firebase={{ logout: () => {} }}
      />
    )).dive();

    const el = component.getElement();
    expect(el.props.children.props.children.length).toBe(8);
    const lastChild = el.props.children.props.children[el.props.children.props.children.length - 1];
    expect(lastChild.type).toBe('div');
    el.props.children.props.children.slice(0, -1).forEach((child) => {
      if (child) {
        expect(['ListItem', 'Divider'].indexOf(child.type.Naked.name) > -1).toBe(true);
        expect(['WithStyles(ListItem)', 'WithStyles(Divider)'].indexOf(child.type.displayName) > -1).toBe(true);
      }
    });
  });

});
