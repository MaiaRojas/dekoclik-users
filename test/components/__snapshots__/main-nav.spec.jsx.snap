// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNav /> should warn when missing nested props: history.push, firebase.logout 1`] = `
Array [
  "Warning: Failed prop type: The prop \`history.push\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav",
]
`;

exports[`<MainNav /> should warn when missing nested props: history.push, firebase.logout 2`] = `
Array [
  "Warning: Failed prop type: The prop \`firebase.logout\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav",
]
`;

exports[`<MainNav /> should warn when missing props: auth, match, history, firebase 1`] = `
Array [
  "Warning: Failed prop type: The prop \`auth\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav",
]
`;

exports[`<MainNav /> should warn when missing props: auth, match, history, firebase 2`] = `
Array [
  "Warning: Failed prop type: The prop \`auth\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav",
]
`;

exports[`<MainNav /> should warn when missing props: auth, match, history, firebase 3`] = `
Array [
  "Warning: Failed prop type: The prop \`auth\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav",
]
`;

exports[`<MainNav /> should warn when missing props: auth, match, history, firebase 4`] = `
Array [
  "Warning: Failed prop type: The prop \`auth\` is marked as required in \`MainNav\`, but its value is \`undefined\`.
    in MainNav",
]
`;
